app/routes/review_offer.js

```diff
@@ -22,11 +23,22 @@ export default AuthorizeRoute.extend({
   },
 
   loadIfAbsent(offerId) {
+    let _this = this;
     let offer = this.get("store").peekRecord("offer", offerId);
+
     if (offer && this.hasLoadedAssociations(offer)) {
       return offer;
     }
-    return this.store.findRecord("offer", offerId, { reload: true });
+
+    return new AjaxPromise(
+      `/offers/${offerId}`,
+      "GET",
+      _this.get("session.authToken"),
+      { exclude_messages: "true" }
+    ).then(function(data) {
+      _this.store.pushPayload(data);
+      return _this.store.peekRecord("offer", offerId);
+    });
```

Comment:
```suggestion
    return new AjaxPromise(
      `/offers/${offerId}`,
      "GET",
      _this.get("session.authToken"),
      { exclude_messages: "true" }
    ).then((data) => {
      this.store.pushPayload(data);
      return this.store.peekRecord("offer", offerId);
    });
```

`let _this = this;` is a very old pre-lambda pattern, which we will never need to ever use again :)

Reply:
also, you can probably use `await` instead of `.then`

Reply:
(none of this is a big deal, just side notes for the future)